src/EA.Weee.Web.Tests.Unit/Controllers/OrganisationRegistrationControllerTests.cs

```diff
@@ -555,5 +557,95 @@ public async Task PostSearchResults_WithSelectedOrganisationId_RedirectsToJoinOr
             Assert.Equal("JoinOrganisation", redirectResult.RouteValues["action"]);
             Assert.Equal(redirectResult.RouteValues["OrganisationId"], new Guid("05DF9AE8-DACE-4173-A227-16933EB5D5F8"));
         }
+
+        [Fact]
+        public async void TonnageTypeGet_ReturnsViewWithViewModel_WithSearchText()
+        {
+            // Arrange
+            var organisationSearcher = A.Dummy<ISearcher<OrganisationSearchResult>>();
+            var weeeClient = A.Dummy<Func<IWeeeClient>>();
+
+            var controller = new OrganisationRegistrationController(
+                weeeClient,
+                organisationSearcher,
+                configurationService);
+
+            const string searchText = "company";
+            var viewModel = new TonnageTypeViewModel()
+            {
+                SearchedText = searchText,
+            };
+
+            // Act
+            var result = await controller.TonnageType(searchText);
+
+            // Assert
+            var resultViewModel = result.Model as TonnageTypeViewModel;
+            Assert.NotNull(result);
+            Assert.Equal(searchText, resultViewModel.SearchedText);
+            Assert.True(string.IsNullOrEmpty(result.ViewName) || result.ViewName == "Tonnage");
+        }
+
+        [Fact]
+        public async void TonnageTypePost_ModelNotValid_ReturnsView()
+        {
+            // Arrange
+            var organisationSearcher = A.Dummy<ISearcher<OrganisationSearchResult>>();
+            var weeeClient = A.Dummy<Func<IWeeeClient>>();
+
+            var controller = new OrganisationRegistrationController(
+                weeeClient,
+                organisationSearcher,
+                configurationService);
+
+            controller.ModelState.AddModelError("error", "error");
+
+            const string searchText = "company";
+            var viewModel = new TonnageTypeViewModel()
+            {
+                SearchedText = searchText
+            };
+
+            // Act
+            var result = await controller.TonnageType(searchText);
```

Comment:
@will-hope98 This doesnt actually execute the POST method as you are passing the searchText to the method it execute the GET method. Update it to pass the viewModel to controller.TonnageType

Reply:
Done